addons/mail/models/mail_message.py

```diff
@@ -571,7 +571,7 @@ def create(self, values_list):
             if 'record_name' not in values and 'default_record_name' not in self.env.context:
                 values['record_name'] = self._get_record_name(values)
 
-            if 'attachment_ids' not in values:
+            if not values.get('attachment_ids'):
                 values['attachment_ids'] = []
```

Comment:
```suggestion
            values.setdefault(attachment_ids, [])
```

Reply:
It is still not the same, as someone giving False explicitly will have errors afterwards, as False would be considered as a valid value for setdefault.